### PR TITLE
Added MoveToObjOnFloor and GraspObjOnFloor, as well as Removed Objects from Skills and into Samplers

### PR DESCRIPTION
**Pull request by wmcclinton**
This PR fixes samplers by making them more sophisticated (like computing relative pose of objects) instead of needing object and state information in skills. Also, importantly we add the ability to pick from the floor and operators + skills needed to do so, which included creating a new floor object and type.

### CODE REVIEW

File: predicators/envs/spot_env.py [hunk 1]
```diff
@@ -764,6 +807,12 @@ def _ontop_classifier(self, state: State,
         is_above_z = (obj_on_pose[2] - obj_surface_pose[2]) > 0.0
         return is_x_same and is_y_same and is_above_z
 
+    def _onfloor_classifier(self, state: State,
+                            objects: Sequence[Object]) -> bool:
+        obj_on, _ = objects
+        assert obj_name_to_apriltag_id.get(obj_on.name) is not None
```
Comment (tomsilver) [inline predicators/envs/spot_env.py:813]: why is this necessary?

File: predicators/spot_utils/spot_utils.py [hunk 2]
```diff
@@ -846,47 +854,28 @@ def hand_movement(self,
...
-        y = hand_y + params[1]
-        z = hand_z + params[2]
+            x = self.hand_x + params[0]  # dx hand
+            y = self.hand_y + params[1]
+            z = self.hand_z + params[2]
+
+        x_cliped = np.clip(x, 0.3, 0.9)
+        y_cliped = np.clip(y, -0.5, 0.5)
+        z_cliped = np.clip(z, 0.2, 0.7)
+        self.relative_move((x - x_cliped), (y - y_cliped), 0.0)
```
Comment (nkumar-bdai) [inline predicators/spot_utils/spot_utils.py:874]: Three things:
1. Add a comment explaining what you're doing here
2. Make the magic number constants you're using be obvious constants at the top of the file
3. (nit) i think 'clipped' is misspelled

File: predicators/ground_truth_models/spot/nsrts.py [hunk 3]
```diff
@@ -39,6 +43,42 @@ def move_sampler(state: State, goal: Set[GroundAtom],
             if len(objs) == 3:
                 if objs[2].name == "extra_room_table":
                     return extra_room_table_offset
+            # For MoveToObjOnFloor
+            if len(objs) == 3:
+                if objs[2].name == "floor":
+                    obj = objs[1]
+                    # Get graph_nav to body frame.
```
Comment (tomsilver) [inline predicators/ground_truth_models/spot/nsrts.py:50]: my hope is that the "state" of the spot interface (like stuff that changes) doesn't actually matter here. is that true? if not, I would be concerned that the spot interface "state" could be misaligned with the `state` passed into the sampler. we should keep all actually-changing stuff inside of the `state`.

File: predicators/ground_truth_models/spot/nsrts.py [hunk 4]
```diff
@@ -39,6 +43,42 @@ def move_sampler(state: State, goal: Set[GroundAtom],
...
+                    gn_state = _spot_interface.get_localized_state()
+                    gn_origin_tform_body = math_helpers.SE3Pose.from_obj(
+                        gn_state.localization.seed_tform_body)
+
+                    # Transform fiducial pose for relative pose.
+                    assert isinstance(obj, Object)
```
Comment (tomsilver) [inline predicators/ground_truth_models/spot/nsrts.py:56]: why is this needed?

File: predicators/spot_utils/spot_utils.py [hunk 5]
```diff
@@ -72,19 +72,8 @@
...
-    402: (6.492544006833252, -5.949555197001736, 0.15848369079640096),
-    403: (9.865172551128433, -6.866927111707811, 0.6057541244901083),
-    404: (6.309634003009213, -5.927127765104969, 0.28815345669215225),
-    405: (7.012502003835815, -8.16002435840359, -0.19144977319953185),
-    406: (9.985505899791097, -6.981086719041378, 0.24004802462770083),
-    407: (6.974322333685671, -8.519032350022558, 0.19594502052006785),
-    408: (6.442939585696927, -6.266242910425788, 0.026100683357255378),
-    409: (8.20282, -6.09703, 0.0314739)
+    "extra_room_table": 409,
+    "floor": 0
```
Comment (tomsilver) [inline predicators/spot_utils/spot_utils.py:76]: this seems weird, can we handle whatever necessitated this another way?

File: predicators/spot_utils/spot_utils.py [hunk 6]
```diff
@@ -72,19 +72,8 @@
     "low_wall_rack": 406,
     "front_tool_room": 407,
     "tool_room_table": 408,
-    "extra_room_table": 409
-}
-
-apriltag_id_to_obj_poses: Dict[int, Tuple[float, float, float]] = {
```
Comment (tomsilver) [inline predicators/spot_utils/spot_utils.py:78]: awesome!

**Review by tomsilver — APPROVED** [review]
not checking the math. looks great otherwise!

File: predicators/envs/spot_env.py (hunk 1 above)
Comment (wmcclinton) [reply predicators/envs/spot_env.py:813]: it's not
